### PR TITLE
Refactor RecommendedFile component and add action buttons

### PR DESCRIPTION
**Pull request by NanoSector**
This fixes https://github.com/nextcloud/server/issues/32370 .

While adding the action button, I refactored the RecommendedFile component to be more flexible. It should now be more responsive at different widths.

In the future, a migration to [DashboardWidgetItem](https://github.com/nextcloud/nextcloud-vue/blob/4a2f796c75f83f93bbb1f31a5abbc9ebce9d28a1/src/components/DashboardWidgetItem/DashboardWidgetItem.vue) might be wanted as it significantly reduces the amount of code required in this application. However, after testing this, that component as of now does not provide support for changing the layout of the presented text or even passing custom HTML to it. Thus we can not show the file extension next to the filename.

CleanShot 2022-05-22 at 16 47 26
CleanShot 2022-05-22 at 16 51 24

### CODE REVIEW

File: src/components/RecommendedFile.vue [hunk 1]
```diff
@@ -150,18 +184,25 @@ export default {
 				window.location = generateUrl('/f/' + this.id)
 			}
 		},
+		navigateToFile() {
+			if (this.isFileListAvailable) {
+				this.changeDirectory(this.directory)
+					.then(() => this.scrollTo(this.name))
+					.catch(console.error.bind(this))
```
Comment (artonge) [inline src/components/RecommendedFile.vue:191]: You can use `async/await` and `try/catch`

Comment (artonge) [inline src/components/RecommendedFile.vue:191]: You can use `async/await` and `try/catch` to make it more readable

File: src/components/Dashboard.vue [hunk 2]
```diff
@@ -67,32 +66,26 @@ export default {
...
+			.recommendation {
+				.thumbnail {
+					width: 44px;
+					height: 44px;
 				}
 
-				.reason {
-					display: none;
+				.details {
+					// 68 = 44 (thumbnail) + 8 (8px right margin)
```
Comment (artonge) [inline src/components/Dashboard.vue:80]: ?

```suggestion
 					// 52 = 44 (thumbnail) + 8 (8px right margin)
```

Comment (artonge) [inline src/components/Dashboard.vue:80]: ?

```suggestion
 					// 52 = 44 (thumbnail) + 8 (8px right margin)
```

**Comment by NanoSector** [comment]
@artonge Thanks for your review, I have addressed your comments.

**Review by artonge — APPROVED** [review]

**Comment by skjnldsv** [comment]
I'm guessing it's too outdated and the new Dashboard API creates some other changes

**Pull request closed without merging**